Foundation/NSTask.swift

```diff
@@ -151,9 +151,27 @@ public class NSTask : NSObject {
     public var currentDirectoryPath: String = NSFileManager.defaultInstance.currentDirectoryPath
     
     // standard I/O channels; could be either an NSFileHandle or an NSPipe
-    public var standardInput: AnyObject?
-    public var standardOutput: AnyObject?
-    public var standardError: AnyObject?
+    public var standardInput: AnyObject? {
+        willSet {
+            guard newValue is NSPipe || newValue is NSFileHandle else {
+                abort()
```

Comment:
Let's change this (and following) to `preconditionFailure("message")` instead, so a caller knows what they did wrong without digging into the source code.